java/core/src/main/java/sleeper/core/statestore/StateStoreProvider.java

```diff
@@ -65,6 +65,28 @@ public StateStore getStateStore(TableProperties tableProperties) {
         return tableIdToStateStoreCache.get(tableId);
     }
 
+    /**
+     * Remove a specific table's state store from the cache.
+     *
+     * @param  tableProperties the Sleeper table properties
+     * @return true if the state store for the requested Sleeper table was in the cache and has been removed
+     */
+    public boolean removeStateStoreFromCache(TableProperties tableProperties) {
+        String tableId = tableProperties.get(TABLE_ID);
+        return this.removeStateStoreFromCache(tableId);
+    }
+
+    /**
+     * Remove a specific table's state store from the cache.
+     *
+     * @param  tableId Sleeper table ID
+     * @return true if the state store for the requested Sleeper table was in the cache and has been removed
+     */
+    public boolean removeStateStoreFromCache(String tableId) {
+        return this.tableIdToStateStoreCache.remove(tableId) != null &&
+            this.tableIds.remove(tableId);
```

Comment:
This could do with unit test coverage in StateStoreProviderTest, ideally through the other method signature to cover both. We could do with a test for a scenario where the table IDs being up to date makes a difference, so that the `tableIds.remove` call is asserted as well.